[TensorExpr] Remove unused classes from TensorExprKernel.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53283 [TensorExpr] Remove unused classes from TensorExprKernel.**

We had `ShapeArg` and `KernelArg` classes, which were wrappers over
`BufferArg` without adding any new functionality on top of what already
existed. This PR removes them and replace their uses with `BufferArg`s
directly.

Differential Revision: [D26821993](https://our.internmc.facebook.com/intern/diff/D26821993)

Commit:
[TensorExpr] Remove unused classes from TensorExprKernel.

We had `ShapeArg` and `KernelArg` classes, which were wrappers over
`BufferArg` without adding any new functionality on top of what already
existed. This PR removes them and replace their uses with `BufferArg`s
directly.

[ghstack-poisoned]

navahgar reviewed (approved):
LGTM 💯 

Any code deletion is great. 👍

Commit:
Update on "[TensorExpr] Remove unused classes from TensorExprKernel."


We had `ShapeArg` and `KernelArg` classes, which were wrappers over
`BufferArg` without adding any new functionality on top of what already
existed. This PR removes them and replace their uses with `BufferArg`s
directly.

Differential Revision: [D26821993](https://our.internmc.facebook.com/intern/diff/D26821993)

[ghstack-poisoned]
